modules/ROOT/pages/dataweave-cookbook-flat-file-read-and-write.adoc

```diff
@@ -0,0 +1,309 @@
+		Type: payload.RQF.Type,
+		"Unique File Identifier": payload.RQF."Unique File Identifier"
+	}
+}
+]]>
+      </ee:set-payload>
+    	</ee:message>
+    </ee:transform>
+    <logger level="INFO" doc:name="Logger" message='#[payload]'/>
+    <file:write doc:name="Write flat file" config-ref="File_Config" path="${fileWritePath}" /> //<2>
+	</flow>
+</mule>
+----
+
+<1> Note that the Transform component specifies the following writer properties in the script header:
++
+* `output`: The output MIME type of the transformation, in this case, `application/flatfile`
+* `schemapath`: The writer property that specifies the path to the flat file schema (`.ffs` file)
+<2> Note that the File Write operation specifies the `path` property to define where to write the flat file
+
+After its execution, the example application generates the following output:
+
+----
+RQH201809011010A000000001USD
+BAT000001HACME RESEARCH                 A000000001
+BAT000000D          0000000000
+BAT000000D          0000000000
+BAT000004T0000120202CR000002A000000001
+BAT000005HAJAX EXPLOSIVES               A000000002
+BAT000000D          0000000000
+BAT000000D          0000000000
+BAT000008T0000003326DB000002A000000002
+RQF0002000008000000116876CRA000000001
+----
+
+== Steps to Test the Examples Locally
+
+If you want to test the examples on your end, follow these steps to create and configure your example Mule application:
+
+. Create a new *Mule Project* in Anypoint Studio.
+. Copy the XML code of the application you want to test into the *Configuration XML* view of your project.
+. xref:studio::add-modules-in-studio-to.adoc[Add the file connector module] to your project.
+. Create a file named `flatfileconfig.properties` in the `src/main.resources` folder of your Mule application and set its content with the following:
++
+----
+fileReadPath=flatfile
+schemaPath=flatfileschema.ffs
+fileWritePath=flatfilenew
+jsonFileReadPath=flatFileJson.json
+fileWorkingDir=#[mule.home ++ "/apps/<your_app_name>/"]
+----
+** Replace `<your_app_name>` with the name of the Mule project you created.
++
+This step sets the working directory for the File connector relative to the `src/main/resources` folder of your Mule application.
+. Create a file named `flatfile` in the `src/main.resources` folder of your Mule application and set its content with the xref:dataweave-flat-file-schemas.adoc#flat-file-sample-data [flat file sample data] from the DataWeave flat file schema example.
+. Create a file named `flatfileschema.ffs` in the `src/main.resources` folder of your Mule application and set its content with the xref:dataweave-flat-file-schemas.adoc#full_schema_ex[full schema example] from the DataWeave flat file schema example.
+. Create a file named `flatFileJson.json` in the `src/main.resources` folder of your Mule application and set its content with the <<json-transformation-output, JSON transformation output>> from the Example of Reading a Flat File.
```

Comment:
```suggestion
. Create a file named `flatFileJson.json` in the `src/main/resources` folder of your Mule application, and set its content with the <<json-transformation-output, JSON transformation output>> from "Example of Reading a Flat File."
```

Reply:
or maybe "... from the example that reads a flat file" instead of "... from the Example of Reading a Flat File." everywhere that occurs